torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -1100,8 +1101,10 @@ std::pair<NvrtcFunction, std::string> nvrtcCompile(
 #ifndef USE_ROCM
   // keeping the string outside the loop for lifetime
   std::string max_register_usage = "--maxrregcount=";
-  uint32_t max_register = 0;
-  if (opt_block_size.has_value() && opt_block_size.value() > 0) {
+  // The maximum possible count allowed by ptxas is 255
+  int max_register = 255;
+  if (max_register_heuristic < 255 ||
```

Comment:
This would fail when `opt-block_size` doesn't have a value. The branch would be taken if `max_register_heuristic` is less than 255 even when `opt_block_size` doesn't have a value, but it is used inside the branch (e.g., at line 1114)

Reply:
nice catch! revised.